tools/discovery/discovery.go

```diff
@@ -0,0 +1,131 @@
+package discovery
+
+import (
+	"bytes"
+	"crypto/tls"
+	"encoding/json"
+	"fmt"
+	"net/http"
+	"net/url"
+	"os"
+
+	"github.com/opencontainers/image-spec/specs-go/v1"
+	"github.com/sirupsen/logrus"
+	"github.com/urfave/cli"
+	"github.com/xiekeyang/oci-discovery/tools/object"
+)
+
+var (
+	defaultTrans = &http.Transport{
+		TLSClientConfig: &tls.Config{InsecureSkipVerify: true},
+	}
+)
+
+func DiscoveryHandler(context *cli.Context) error {
+	var (
+		name  = context.Args()[0]
```

Comment:
The Python code currently [iterates over possibly many names][1], which is why it [writes a name-keyed object][2].  That allows you to save redundant ref-engine discovery calls if several names use the same `host` (although the current Python implementation doesn't take advantage of this possible optimization).  I think it's a useful pattern to follow for the Go tools.

If, on the other hand, you decide to only ever resolve one name at a time, you should be raising usage errors if [`context.NArg()`][3] is not one.

[1]: https://github.com/xiekeyang/oci-discovery/blob/44ec3cf3113e29a743ad04220ccb7ff5197dab2a/oci_discovery/ref_engine_discovery/__main__.py#L67
[2]: https://github.com/xiekeyang/oci-discovery/blame/44ec3cf3113e29a743ad04220ccb7ff5197dab2a/README.md#L43-L66
[3]: https://godoc.org/github.com/urfave/cli#Context.NArg